nova/virt/vmwareapi/volumeops.py

```diff
@@ -704,6 +704,7 @@ def _detach_volume_fcd(self, connection_info, instance):
                 raise exception.Invalid(_('%s does not support disk '
                                           'hotplug.') % adapter_type)
         volume_uuid = connection_info['volume_id']
+        data['volume_id'] = volume_uuid
         device = self._get_vmdk_backed_disk_device(vm_ref, data)
```

Comment:
Why does the other code not need to add `volume_id` to `data`? Is there different stuff returned from Cinder for FCD vs. VMDK?

Reply:
Yes, the original intention was to just add the fcd_obj_id and ds_mo_ref.
It was also a surprise for me, and then i checked.
https://github.com/sapcc/cinder/blob/sap_fcd/cinder/volume/drivers/vmware/fcd.py#L190-L196
Old driver did also not set the cinder uuid on the backing.uuid, so it was logical that they can't use the volume_id.
In our variant, this is no longer the case, the cinder id equals the backing.uuid

Reply:
We can also modify cinder to pass the volume_id in the data

Reply:
Also they originally were thinking, that we would run nova-manage to refresh bdm info, that looks too manually for me

Reply:
Yeah, I would prefer to have Cinder return the value where we need/expect it. But we can keep this fix for now. A comment would be great though :)